### PR TITLE
Back `Stokes` map with a single dense array

### PR DESCRIPTION
**Pull request by sbiquard**
Refactors Stokes/StokesI/StokesQU/StokesIQU/StokesIQUV to be backed by one dense array with the Stokes components on the leading axis (`(n_stokes, *shape)`), instead of one array field per component.

Why: makes the Stokes axis a real, batchable array axis instead of a fixed pytree shape, so Stokes-axis operations (QU rotation, per-leg noise, SHT/beam batching, component mixing) become plain array ops instead of per-leaf pytree.map. Also the smallest concrete instance of issue #18's `TreeArray` idea (fixed treedef = Stokes labels, one materialized axis).

Updated for the new leading axis: pointing (on-the-fly sampling/binning), beam/SHT operators (batch over the Stokes axis via jax-healpy's map2alm/alm2map), SED operators, mapmaker (mask/noise model/hits/preconditioner), results saving. Generic axis-0 operators (IndexOperator, RavelOperator) at Stokes-map call sites use an `(..., idx)` / offset convention to skip the leading axis.

Demodulated TOD mapmaking (single-array Stokes TOD) is adapted as part of this: noise model/operator, mask, hits computation.

Followed by a cleanup pass (reuse/simplification/efficiency review): removes two moveaxis calls from the preconditioner's hot CG path in favor of a single einsum, collapses the ten Stokes arithmetic dunders into one helper, and a few smaller dedup/no-op fixes.

### CODE REVIEW

File: tests/math/test_sht.py [hunk 1]
```diff
@@ -117,10 +107,11 @@ def test_inverse_in_structure_matches_out_structure(self, map2alm):
         """Inverse operator's in_structure must equal Map2Alm's out_structure."""
         assert map2alm.I.in_structure == map2alm.out_structure
 
+    @pytest.mark.skip(reason='upstream jax-healpy bug')
```
Comment (sbiquard) [inline tests/math/test_sht.py:112]: This is tracked in https://github.com/CMBSciPol/jax-healpy/issues/17

File: src/furax/obs/operators/_seds.py [hunk 2]
```diff
@@ -68,20 +68,30 @@ class AbstractSEDOperator(BroadcastDiagonalOperator):
...
+
+    def mv(self, x: PyTree[Inexact[Array, '...']]) -> PyTree[Inexact[Array, '...']]:
+        # Insert the frequency axis before the trailing spatial axis, unless already present
+        # (e.g. a genuine multi-frequency map passed in directly).
+        def func(leaf: Inexact[Array, '...']) -> Inexact[Array, '...']:
+            if leaf.ndim < self.frequencies.ndim:
+                leaf = jnp.expand_dims(leaf, axis=-2)
+            return self.sed() * leaf
+
+        return jax.tree.map(func, x)
```
Comment (sbiquard) [inline src/furax/obs/operators/_seds.py:97]: This new `mv()` is needed because the broadcasting frequency axis isn't anymore the leading dimension in the Stokes container. This is not handled by `BroadcastDiagonalOperator` which was the parent class.
